packages/host/app/components/operator-mode/code-mode.gts

```diff
@@ -392,28 +396,102 @@ export default class CodeMode extends Component<Signature> {
+                      // here we check if a field of an exported card or field is a field of a local one
+                      if (value.possibleFields) {
+                        for (let [fieldName, v] of value.possibleFields) {
+                          if (isInternalReference(v.card)) {
+                            const indexOfParentField = v.card.classIndex;
+                            if (indexOfParentField !== undefined) {
+                              const parentFieldClass =
+                                possibleCardsOrFields[indexOfParentField];
+                              let localName = parentFieldClass.localName;
+                              if (localName) {
+                                let field = getField(cardOrField, fieldName);
+                                if (field && field.card) {
+                                  localParentsOfExportedCardsOrFields.set(
+                                    parentFieldClass,
+                                    field.card,
+                                  );
+                                }
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
                   return {
                     ...value,
-                    cardType: getCardType(
-                      this,
-                      () => cardOrField as typeof BaseDef,
-                    ),
                     cardOrField,
-                  } as CardOrField & Partial<PossibleCardOrFieldClass>;
+                  };
                 }
-                return value as BaseDeclaration;
+                return value;
               },
-            );
+            ) as ({
+              cardOrField: typeof BaseDef;
+            } & PossibleCardOrFieldClass)[];
+
+            // This loop
+            // - adds card type (not loaded tho)
+            // - includes card or field that was local but exported thru some relationship
+            elements = elementsNoCardType.map((value) => {
+              if (value.cardOrField) {
+                return {
+                  ...value,
+                  cardType: getCardType(
+                    this,
+                    () => value.cardOrField as typeof BaseDef,
+                  ),
+                } as CardOrField & Partial<PossibleCardOrFieldClass>;
+              } else {
+                if (localParentsOfExportedCardsOrFields.has(value)) {
+                  let cardOrField = localParentsOfExportedCardsOrFields.get(
+                    value,
+                  ) as typeof BaseDef;
+                  return {
+                    ...value,
+                    cardOrField,
+                    cardType: getCardType(this, () => cardOrField),
+                  } as CardOrField & Partial<PossibleCardOrFieldClass>;
+                }
+              }
+              return value as BaseDeclaration;
+            });
```

Comment:
The purpose of this logic here is to really consolidate the information that we obtain from 
- moduleSyntax. The elements derive using the babel plugin
- importedModule. The module resource that collects exports of cards or fields

The point of it is that we can't just simply know that an element is a card or not from moduleSyntax. Rather, we have to look into the relationships of the cards or fields received from the importedModule. In doing so, we can infer about other cards or fields that are defined locally but not exported.

Reply:
Ive refactored this so its more readable